### PR TITLE
PR: Improve UX of the Plots plugin

### LINKED ISSUE #12158: Move plots in Plots pane
Possible? I currently have to re-plot stuff to order them a certain way for viewing, which can be quite expensive. The Plots pane is nice for directly comparing images, but this feature seems absent.

### LINKED ISSUE #21899: Feature Request: Be able to double click plots so they blow up into a larger screen
So from other IDEs I had the tendency to click/double click the picture for it to become bigger. Would be nice to have this future in Spyder as well

### PR DESCRIPTION
**Pull request by ccordoba12**
## Description of Changes

- Move `Fit plots` action to the plugin toolbar and make plots be fitted automatically to the pane the first time they are generated. That makes the zoom in/out buttons be enabled by default, which is a much better usability experience than having them disabled.
- Disabling `Fit plots` shows them at full size.
- Save the `Fit plots` and zoom level for every plot. That way, we'll be able to show each plot in the specific way users want to see them.
- Implement drag and drop of thumbnails so that users can manually reorganize plots as they see fit. 
- Improve organization of buttons in its main toolbar and increase default width of thumbnail bar to go with this change.
- Make doing double click on a plot show it at full size.

### Visual changes

* Main toolbar
    - Before 
        imagen
    
    - After
        imagen

* `Fit plots` button

    fit-button

* Separate zoom level for each plot

    zoom-per-plot

* Drag and drop

    plot-drag-and-drop

* Zoom in on double click

    plot-double-click

### Issue(s) Resolved

Fixes #21899.
Fixes #12158
Part of https://github.com/spyder-ide/ux-improvements/issues/19

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, could you review if functionality of these changes is working as expected? Thanks!

About the UI/UX changes, I already discussed with the UX team and they agreed with them.

**Comment by dalthviz** [comment]
Gave this a check and seems like the double click is changing the zoom factor value even when the option to `Fit plots to window` is enabled causing a sudden zoom factor change when the option gets disabled again. Probably the zoom factor when disabling the `Fit plots to window` should be the one the plot have before the option was enabled but not sure if that is intended. Here a preview of that behavior:

plots_double_click

Other than that seems like things are working :+1:

**Comment by ccordoba12** [comment]
Great catch @dalthviz! I disabled doing zooming in and out with double clicks when `Fit plots to window` is enabled. That actually doesn't make much sense because the zoom factor is fixed to the size of the Plots pane.

**Comment by dalthviz** [comment]
Gave another check and although the click with the option enabled is fixed, seems like now (or at least I didn't noticed that before 🤔) a double double click needs to be done to zoom up to 100%:

plots_double_double_click

I think is related with passing to another plot when the previous one was left at 100% 🤔

**Comment by ccordoba12** [comment]
Thanks for the additional review @dalthviz! After considering it carefully, I decided to push the following improvements to make the UX more intuitive:

* Move the fit action to the main toolbar as a toggle and check it after every a new plot is generated.
* When that action is unchecked the plot is shown at full size.
* When the user does zoom in/out on a plot, that action is automatically unchecked.
* Preserve the zoom level of each plot so that switching plots in the thumbnail bar shows them exactly as users were seeing them the last time. 

I updated the OP with these new changes accordingly.

**Comment by dalthviz** [comment]
Gave this another check and seems like now, besides the previous behavior (double double click need to effectivly change between zoom factors when changing between plots), there is also a possiblitity to make the zoom factor to get stuck in 100% 🤔:

double_double_click_zoom_2

**Comment by ccordoba12** [comment]
@dalthviz, I didn't understand that problem but I do now. So, I rewrote one of my previous commits to fix it. That removed the ability to zoom out to the previous size with a second double-click. But I think that's not so important now that I added the auto-fit button to the toolbar.

So, please check again.

**Comment by dalthviz** [comment]
Checked again and now I'm unable to reproduce the mentioned behaviors so seems like things are working as described :+1: The only suggestion that came to mind while checking the new behavior is to maybe add a shortcut for the new auto-fit button.

File: spyder/plugins/plots/widgets/figurebrowser.py [hunk 1]
```diff
@@ -423,13 +454,42 @@ def show_context_menu(self, qpoint):
...
+            lambda: self.verticalScrollBar().setValue(
+                self.current_thumbnail.vscrollbar_value
+            ),
+        )
+        QTimer.singleShot(
+            20,
+            lambda: self.horizontalScrollBar().setValue(
+                self.current_thumbnail.hscrollbar_value
+            ),
+        )
```
Comment (dalthviz) [inline spyder/plugins/plots/widgets/figurebrowser.py:491]: Maybe an explanation of why these timers need to be set here could be worthy

**Comment by ccordoba12** [comment]
> The only suggestion that came to mind while checking the new behavior is to maybe add a shortcut for the new auto-fit button.

Good idea. I set `Ctrl+0` for this action.

File: spyder/plugins/plots/widgets/figurebrowser.py (hunk 1 above)
Comment (ccordoba12) [reply spyder/plugins/plots/widgets/figurebrowser.py:493]: Done at the top of these timers.

**Review by dalthviz — APPROVED** [review]
Thanks @ccordoba12 ! Things LGTM :+1: Maybe adding some sort of test could be nice to have but leaving this approved so feel free to merge

**Comment by ccordoba12** [comment]
I added a couple of tests to cover the new auto-fit and zoom level per plot functionality, which I think are the most important ones. So, this is ready now.

**Pull request merged**